src/avro/src/schema.rs

```diff
@@ -297,17 +297,14 @@ pub enum SchemaPiece {
 impl SchemaPiece {
     /// Returns whether the schema node is "underlyingly" an Int (but possibly a logicalType typedef)
     pub fn is_underlying_int(&self) -> bool {
-        match self {
-            SchemaPiece::Int | SchemaPiece::Date => true,
-            _ => false,
-        }
+        matches!(self, SchemaPiece::Int | SchemaPiece::Date)
```

Comment:
Replacing simple language syntax with some macro? I would much rather just turn off the lint

Reply:
like, this strikes me as the same as if in C someone came out with
```
#define BACKWARDS_FOR(x, y, z) for (z; y; x)
```
and then made a lint asking people to use it. Like, yes, it works, but... why do it?

Reply:
fwiw I like `matches!`, but have no strong opinion.

Reply:
I like it slightly more too. But if folks hate it I'm definitely on board with disabling the lint. Just went with the clippy default here out of inertia, not because I have strong opinions either way.